OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace OJS.Workers.ExecutionStrategies.Extensions
+{
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+
+    using Ionic.Zip;
+
+    using OJS.Workers.Common;
+
+    public static class ExecutionContextExtensions
+    {
+        public static void SanitizeContent(this ExecutionContext executionContext)
+        {
+            var callingClassName = new StackFrame(1).GetMethod()?.DeclaringType?.Name;
+
+            switch (callingClassName)
+            {
+                case nameof(DotNetCoreProjectTestsExecutionStrategy):
+                case nameof(DotNetCoreProjectExecutionStrategy):
+                case nameof(DotNetCoreUnitTestsExecutionStrategy):
+                case nameof(DotNetCoreTestRunnerExecutionStrategy):
+                    SanitizeDotNetCoreZipFile(executionContext);
+                    break;
+            }
+        }
+
+        private static void SanitizeDotNetCoreZipFile(ExecutionContext executionContext)
+        {
+            if (!ExecutionContextContainsZipFile(executionContext))
+            {
+                return;
+            }
+
+            const string connectionStringSearchPattern = @"(\.\s*UseSqlServer\s*\()(.*)(\))";
```

Comment:
Extract into a constant elsewhere. Some global constant file or at least at the top of the declaring class